clients/src/test/java/org/apache/kafka/common/requests/FetchRequestTest.java

```diff
@@ -243,4 +244,45 @@ public void testPartitionDataEquals() {
             new FetchRequest.PartitionData(Uuid.randomUuid(), 300, 0L, 300, Optional.of(300)));
     }
 
+    @ParameterizedTest
+    @MethodSource("fetchVersions")
+    public void testFetchRequestNoCacheData(short version) {
+        boolean fetchRequestUsesTopicIds = version >= 13;
+        Uuid topicId = Uuid.randomUuid();
+        int partition = 0;
+        TopicIdPartition tp = new TopicIdPartition(topicId, partition, "topic");
+
+        Map<TopicPartition, FetchRequest.PartitionData> partitionData = Collections.singletonMap(tp.topicPartition(),
+                new FetchRequest.PartitionData(topicId, 0, 0, 0, Optional.empty()));
+        List<TopicIdPartition> toReplace = Collections.singletonList(tp);
+
+        FetchRequest fetchRequest = FetchRequest.Builder
+                .forReplica(version, 0, 1, 1, 1, partitionData)
+                .removed(Collections.emptyList())
```

Comment:
Could you add some "removed" TopicIdPartition? That can avoid empty `forgottenTopicMap`

Reply:
I according to diffenet version to generate fetchRequest, that can avoid empty `forgottenTopicMap`